pkg/operator/ceph/object/config.go

```diff
@@ -65,7 +65,7 @@ func (c *clusterConfig) portString() string {
 
 	port := c.store.Spec.Gateway.Port
 	if port != 0 {
-		if !c.clusterSpec.Network.IsHost() {
+		if !c.clusterSpec.Network.IsHost() && c.store.Spec.HostNetwork == nil || c.store.Spec.HostNetwork != nil && !*c.store.Spec.HostNetwork {
```

Comment:
the priority needs to give to c.store.Spec.HostNetwork otherwise `IsHost()` so the condition should be
if (( c.store.Spec.HostNetwork != nil && !*c.store.Spec.HostNetwork ) || (c.store.Spec.HostNetwork == nil && c.clusterSpec.Network.IsHost()))

IMO it is better to do this as use the helper function in `ObjectStoreSpec` then use that api in both places: like
```
func (s *ObjectStoreSpec) IsHostNetwork(c *clusterSpec) bool {
if Spec.HostNetwork != nil {
  return Spec.HostNetwork 
}
return c.Network.IsHost()
}
```

Reply:
good suggestion.